src/formats/onenote.ts

```diff
@@ -0,0 +1,444 @@
+import { FormatImporter } from 'format-importer';
+import { ProgressReporter } from 'main';
+import { DataWriteOptions, Notice, Setting, TFile, TFolder, htmlToMarkdown } from 'obsidian';
+import { parseHTML } from '../util';
+import { MicrosoftGraphHelper } from './onenote/graph-helper';
+import { OnenotePage, OnenoteSection, Notebook, SectionGroup, User, FileAttachment } from '@microsoft/microsoft-graph-types';
+
+// TODO: This array is used by a few other importers, so it could get moved into format-importer.ts to prevent duplication
+const ATTACHMENT_EXTS = ['png', 'webp', 'jpg', 'jpeg', 'gif', 'bmp', 'svg', 'mpg', 'm4a', 'webm', 'wav', 'ogv', '3gp', 'mov', 'mp4', 'mkv', 'pdf'];
+
+export class OneNoteImporter extends FormatImporter {
+	useDefaultAttachmentFolder: boolean = true;
+	importIncompatibleAttachments: boolean = false;
+	// In the future, enabling this option will only import InkML files.
+	// It would be useful for existing OneNote imports or users whose notes are mainly drawings.
+	importDrawingsOnly: boolean = false;
+	microsoftAccountSetting: Setting;
+	contentArea: HTMLDivElement;
+
+	attachmentQueue: FileAttachment[] = [];
+	selectedSections: OnenoteSection[] = [];
+
+	graphHelper: MicrosoftGraphHelper = new MicrosoftGraphHelper();
+
+	init() {
+		this.addOutputLocationSetting('OneNote');
+		this.showUI();
+		// Required for the OAuth sign in flow
+		this.modal.plugin.registerObsidianProtocolHandler('importer-onenote-signin', (data) => {
```

Comment:
There's currently an issue if you:

1. Open the importer UI
2. Select OneNote as the file format
3. Change the file format to something else
4. Change it back to OneNote

Note the error in the console, as it attempts to register the protocolHandler again even though it has already been registered once.

One solution to this would be to use the `unregister` callback that is provided when calling `register`, although I am not sure when it should be called.

@lishid or @ericaxu do you have any suggestions?

Reply:
Clicking "Sign in" also does not work after following those steps.